switch_mod/solve.py

```diff
@@ -526,28 +559,18 @@ def solve(model):
         solve_end_time = time.time()
         print "Solved model. Total time spent in solver: {:2f} s.".format(solve_end_time - solve_start_time)
     
+    # check for errors
     model.solutions.load_from(results)
-
-    # Only return if the model solved correctly, otherwise throw a useful error
-    if(results.solver.status == SolverStatus.ok and 
-       results.solver.termination_condition == TerminationCondition.optimal):
-        return results
-    elif (results.solver.termination_condition == TerminationCondition.infeasible):
+    if results.solver.termination_condition == pyomo.opt.TerminationCondition.infeasible:
         if hasattr(model, "iis"):
             print "Model was infeasible; irreducible infeasible set (IIS) returned by solver:"
             print "\n".join(c.cname() for c in model.iis)
         else:
             print "Model was infeasible; if the solver can generate an irreducible infeasible set,"
             print "more information may be available by calling this script with --suffixes iis ..."
         raise RuntimeError("Infeasible model")
-    else:
```

Comment:
I have no idea why this section got deleted. I may have done it accidentally. Reverting it.